File: c/005.c
#include <stdio.h>

int lcm(int, int);
int gcd(int, int);

int main() {
    int ans = 1;

    for (int i = 2; i <= 20; i++) {
        ans = lcm(ans, i);
    }

    printf("%d\n", ans);
    return 0;
}

// we don't have to worry about
// negative values of a and b here
int lcm(int a, int b) {
    return a / gcd(a, b) * b;  // prevent overflow
}

int gcd(int a, int b) {
    return b == 0 ? a : gcd(b, a % b);
}

File: golang/005.go
package main

import "fmt"

func abs(n int) int {
	if n < 0 {
		return -n
	} else {
		return n
	}
}

func gcd(a, b int) int {
	if b == 0 {
		return a
	} else {
		return gcd(b, a%b)
	}
}

func lcm(a, b int) int {
	return abs(a*b) / gcd(a, b)
}

func main() {
	ans := 1
	for i := 1; i <= 20; i++ {
		ans = lcm(ans, i)
	}
	fmt.Println(ans)
}
